### PR TITLE
Cherrypick 0.14.x

### PR DESCRIPTION
**Pull request by akelad**
**Proposed changes**:
- more changes to cherrypick...

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: rasa_nlu/evaluate.py [hunk 1]
```diff
@@ -84,9 +86,8 @@ def plot_confusion_matrix(cm,
                           title='Confusion matrix',
                           cmap=None,
                           zmin=1,
-                          out=None) -> None:  # pragma: no cover
+                          out=None):  # pragma: no cover
```
Comment (MetcalfeTom) [inline rasa_nlu/evaluate.py]:
```suggestion
                          out=None) -> None:  # pragma: no cover
```

File: rasa_nlu/evaluate.py [hunk 2]
```diff
@@ -125,9 +126,9 @@ def plot_confusion_matrix(cm,
         fig.savefig(out, bbox_inches='tight')
 
 
-def plot_histogram(hist_data: List[List[float]],
-                   out: Optional[Text] = None
-                   ) -> None:  # pragma: no cover
+def plot_histogram(hist_data,  # type: List[List[float]]
+                   out=None  # type: Optional[Text]
+                   ):  # pragma: no cover
```
Comment (MetcalfeTom) [inline rasa_nlu/evaluate.py]: Revert this

File: rasa_nlu/evaluate.py [hunk 3]
```diff
@@ -152,11 +153,11 @@ def plot_histogram(hist_data: List[List[float]],
...
-def log_evaluation_table(report: Text,
-                         precision: float,
-                         f1: float,
-                         accuracy: float
-                         ) -> None:  # pragma: no cover
+def log_evaluation_table(report,  # type: Text
+                         precision,  # type: float
+                         f1,  # type: float
+                         accuracy  # type: float
+                         ):  # pragma: no cover
```
Comment (MetcalfeTom) [inline rasa_nlu/evaluate.py]: Revert these

File: rasa_nlu/evaluate.py [hunk 4]
```diff
@@ -203,7 +204,8 @@ def clean_intent_labels(labels):
     return [l if l is not None else "" for l in labels]
 
 
-def drop_intents_below_freq(td: TrainingData, cutoff: int = 5):
+def drop_intents_below_freq(td, cutoff=5):
+    # type: (TrainingData, int) -> TrainingData
```
Comment (MetcalfeTom) [inline rasa_nlu/evaluate.py]: Revert these

File: rasa_nlu/evaluate.py (hunk 2 above)
Comment (akelad) [reply rasa_nlu/evaluate.py]: yes, please do that :P

Comment (akelad) [reply rasa_nlu/evaluate.py]: you can do it with the suggestion function and then commit them all together

**Comment by akelad** [comment]
seems to look good now, let's pray this builds 🙏

**Pull request merged**